doc: Add release notes for 2.5 (#12697)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.5.0.rst

The following is the most important change inherited from the upstream Matter:

* Added the :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT` Kconfig option to manage watermark support.

* Updated:

  * The factory data guide with an additional rotating ID information.
  * Set onboarding code generation to be enabled by default if the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` Kconfig is set.

* Fixed RAM and ROM reports.

Wi-Fi®
------

* Added:

  * Integration of Wi-Fi connectivity with Connection Manager connectivity API.
  * The :kconfig:option:`CONFIG_NRF_WIFI_IF_AUTO_START` Kconfig option to enable an application to set or unset ``AUTO_START`` on an interface.
    This can be done by using the ``NET_IF_NO_AUTO_START`` flag.
  * Support for sending TWT sleep/wake events to applications.
  * The nRF5340 HFCLK192M clock divider is set to the default value ``Div4`` for lower power consumption when the QSPI peripheral is idle.
  * Extensions to the scan command to provide better control over some scan parameters.

Applications
============

This section provides detailed lists of changes by :ref:`application <applications>`.

> Added new application :ref:Matter bridge <matter_bridge_app> that provides support for the following:

Author: Good catch, thanks Marek. Fixed.

  * Support for bridging of the Bluetooth LE Environmental Sensor (ESP).
  * Support for performing Device Firmware Upgrade (DFU) over Bluetooth LE using Simple Management Protocol (SMP).

Samples
=======

Bluetooth samples
-----------------

* :ref:`direct_test_mode` sample:

  * Added:

    * Support for the nRF52840 DK.
    * Experimental support for the HCI interface.

  * Updated:

    * Aligned timers' configurations to the new nrfx API.
    * Extracted the DTM radio API from the transport layer.
    * Added support for the radio fast ramp-up feature.
      This feature is enabled by default.

* :ref:`peripheral_hids_keyboard` sample:

  * Fixed an interoperability issue with iOS devices by setting the report IDs of HID input and output reports to zero.

* :ref:`fast_pair_input_device` sample:

  * Renamed the sample from Bluetooth: Fast Pair to :ref:`fast_pair_input_device` and moved it to the :file:`samples/bluetooth/fast_pair` folder.
  * Added automatic switching to the Fast Pair not discoverable advertising mode with the hide UI indication instead of removing the Fast Pair advertising payload when all bond slots are taken.

>   * Increased the system workqueue stack size (:kconfig:option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE`) to ``1200`` to prevent stack overflows right after booting the nRF5340 DK.

Reviewer: Ok, merged already :)

  * :ref:`wifi_shutdown_sample` sample that demonstrates how to configure the Wi-Fi driver to shut down the Wi-Fi hardware when the Wi-Fi interface is not in use.
  * :ref:`wifi_twt_sample` sample that demonstrates how to establish TWT flow and transfer data conserving power.
  * Support for the Wi-Fi driver to several upstream Zephyr networking samples.
  * Enhancements to support device re-trimming process in the :ref:`wifi_radio_test` sample.
  * Changes to the :ref:`wifi_scan_sample` sample to demonstrate usage of new scan APIs.

Other samples
-------------

* Added the :ref:`802154_sniffer` sample.

* Removed the Random hardware unique key sample.
  The sample is redundant since its functionality is presented as part of the :ref:`hw_unique_key_usage` sample.

* :ref:`radio_test` sample:

  * Updated the sample by aligning the timer's configuration to the new nrfx API.

Drivers
=======

This section provides detailed lists of changes by :ref:`driver <drivers>`.

* Added :ref:`bme68x_iaq` to run the Bosch Sensor Environmental Cluster (BSEC) library in order to get Indoor Air Quality (IAQ) readings.

Wi-Fi drivers
-------------

> 

Reviewer: remove one empty line

Author: Fixed.

  Snippet is used for nRF91 modem tracing with the UART backend for the following applications and samples:

  * :ref:`asset_tracker_v2`
  * :ref:`serial_lte_modem`
  * All samples that use nRF91 Series DK except for :ref:`slm_shell_sample`, :ref:`modem_trace_flash`, :ref:`modem_trace_backend_sample`.

  For samples where the UART trace backend is enabled by default, the configuration is added to the sample overlays and project configuration.

* The default board revision for nRF9160 DK has changed to v0.14.0.
  See :ref:`nrf9160_board_revisions` for more details.

Working with nRF52 Series
=========================

* :ref:`ug_nrf52_developing`:

  * Updated the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option to support MCUboot bootloader in the direct-xip mode, and added the related documentation.
    For details, see :ref:`mcuboot_direct_xip_mode`.

Protocols
=========

This section provides detailed lists of changes by :ref:`protocol <protocols>`.
See `Samples`_ for lists of changes for the protocol-related samples.

Amazon Sidewalk
---------------

Starting from 2.5.0 release, Amazon Sidewalk is a part of the |NCS|.

* Added:

>   * nRF53 production support for Amazon Sidewalk, including Bluetooth LE, LoRa, and FSK.

Reviewer 2: ```suggestion
  * nRF53 production support for Amazon Sidewalk, including Bluetooth® LE, LoRa, and FSK.
```

  * New distribution model (Amazon Sidewalk is now merged with nRF Connect SDK).

* Updated:

  * Suspended external flash in the Sidewalk mode to reduce power consumption.
    External flash is only available in the DFU mode.
  * Merged Amazon Sidewalk libraries into a unified library.
  * Adjusted the sensor monitoring app to change to initialization state upon reconnecting.
    This setting prevents the cloud application from being stuck when it is disconnected for a long time.
  * Deactivated the Bluetooth LE GATT Client to optimize the configuration.

* Removed an experimental devcontainer for a better user experience.

> Bluetooth® LE

Reviewer 2: ```suggestion
Bluetooth LE
```

* :ref:`nrf_cloud_multi_service` sample:

  * Renamed Cellular: nRF Cloud MQTT multi-service to :ref:`nrf_cloud_multi_service`.
  * Added:

    * Documentation for using the :ref:`lib_nrf_cloud_alert` and :ref:`lib_nrf_cloud_log` libraries.
    * The :file:`overlay_coap.conf` file and made changes to the sample to enable the use of CoAP instead of MQTT to connect with nRF Cloud.
    * An overlay that allows the sample to be used with Wi-Fi instead of LTE (MQTT only).
    * Reporting of device and connection info to the device shadow.
    * The :file:`overlay_min_coap.conf` and :file:`overlay_min_mqtt.conf` overlay files.
    * Handling of shadow deltas caused by alert and log configuration changes for CoAP.

  * Updated:

    * The :file:`overlay_nrfcloud_logging.conf` file to enable JSON logs by default.
    * The encoding and decoding of nRF Cloud data to use the :c:struct:`nrf_cloud_obj` structure and associated functions.
    * The connection logic by cleaning and simplifying it.
    * The sample to use Zephyr's ``conn_mgr`` and the :kconfig:option:`CONFIG_LTE_CONNECTIVITY` Kconfig option instead of using the :ref:`lte_lc_readme` library directly.

>     * The sample to remove redundant shadow updates for nRF Cloud.Legitimate server side CoAP API errors

Reviewer 2: ```suggestion
    * The sample to remove redundant shadow updates for nRF Cloud Legitimate server side CoAP API errors.
```

    * Header files for buttons and LEDs definition required by the :ref:`lib_caf` library.
    * An :file:`overlay-nrf_provisioning.conf` file to enable the :ref:`lib_nrf_provisioning` library.

  * Updated:

    * The sample to use the :ref:`lib_caf` library instead of the :ref:`dk_buttons_and_leds_readme` library.
    * The sample now displays an error message when it fails to send an alert to nRF Cloud.

* :ref:`udp` sample:

   * Updated the sample to use the Kconfig option :kconfig:option:`CONFIG_LTE_RAI_REQ` and socket options ``SO_RAI_*`` for Release Assistance Indication (RAI) functionality.

Thread samples
--------------

* Removed support for the ``nrf52833dk_nrf52833`` build target in the :ref:`ot_cli_sample`, :ref:`coap_client_sample`, and :ref:`coap_server_sample` samples.
* Updated the build target ``nrf52840dongle_nrf52840`` to use USB CDC ACM as serial transport as default.
  Samples for this target can now be built without providing extra configuration arguments.

Matter samples
--------------

* Added the :ref:`Matter thermostat <matter_thermostat_sample>` sample.

* Updated:

>   * Matter over Thread samples by disabling OpenThread shell by default .

Reviewer 2: ```suggestion
  * Matter over Thread samples by disabling OpenThread shell by default.
```

    Their functionality is merged into the ``#XGPS`` AT command that now allows using A-GNSS and P-GPS at the same time.
  * The AT command ``#XSLMUART``.
    UART is now configured using only devicetree.

nRF5340 Audio
-------------

* The whole application architecture for handling Bluetooth LE Audio has changed. New modules have been added:

  * Management - This module handles scanning and advertising, in addition to general initialization, controller configuration, and transfer of DFU images.
  * Stream - This module handles the setup and transfer of audio in the Bluetooth LE Audio context.
  * Renderer - This module handles rendering, such as volume up and down.
  * Content Control - This module handles content control, such as play and pause.

* Added back the QDID number for the :ref:`lib_bt_ll_acs_nrf53_readme` to the documentation.
* Updated the :ref:`application documentation <nrf53_audio_app>` by splitting it into several pages.

nRF Machine Learning (Edge Impulse)
-----------------------------------

* Updated the machine learning models (:kconfig:option:`CONFIG_EDGE_IMPULSE_URI`) used by the application so that they are now hosted by Nordic Semiconductor.

> nRF Desktop

Reviewer 3: @MarekPieta, please also take a look at the nRF Desktop section

  * To improve the accuracy, the generation of simulated movement data in the :ref:`nrf_desktop_motion` now uses a timestamp in microseconds based on the cycle count (either :c:func:`k_cycle_get_32` or :c:func:`k_cycle_get_64` function depending on the :kconfig:option:`CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER` Kconfig option).
  * Aligned Kconfig option names in the :ref:`nrf_desktop_motion` implementation that generates motion from button presses.
    The Kconfig options defining used key IDs are prefixed with ``CONFIG_MOTION_BUTTONS_`` instead of ``CONFIG_MOTION_`` to ensure consistency with configuration of other implementations of the motion module.
  * The :ref:`nrf_desktop_ble_scan` no longer stops Bluetooth LE scanning when it receives :c:struct:`hid_report_event` related to a HID output report.
    Sending HID output report is triggered by a HID host.
    Scanning stop may lead to an edge case where the scanning is stopped, but there are no peripherals connected to the dongle.

Thingy:53: Matter weather station
---------------------------------

* Added support for the nRF7002 Wi-Fi expansion board.

Matter Bridge
-------------

> * Added:

Author: Done.

  * The :ref:`Matter bridge <matter_bridge_app>` application.
  * Support for the Bluetooth LE bridged devices.
  * Support for bridging of the Bluetooth LE Environmental Sensor (ESP).
  * Support for performing Device Firmware Upgrade (DFU) over Bluetooth LE using Simple Management Protocol (SMP).

Samples
=======

Bluetooth samples
-----------------

* :ref:`direct_test_mode` sample:

  * Added:

    * Support for the nRF52840 DK.
    * Experimental support for the HCI interface.

  * Updated:

    * Aligned timers' configurations to the new nrfx API.
    * Extracted the DTM radio API from the transport layer.
    * Added support for the radio fast ramp-up feature.
      This feature is enabled by default.

* :ref:`peripheral_hids_keyboard` sample:

  * Fixed an interoperability issue with iOS devices by setting the report IDs of HID input and output reports to zero.

> * :ref:`fast_pair_input_device` sample:

Reviewer 3: @alstrzebonski, please also take a look at the Fast Pair sample section in NCS 2.5.0 release notes

Drivers
=======

This section provides detailed lists of changes by :ref:`driver <drivers>`.

* Added :ref:`bme68x_iaq` to run the Bosch Sensor Environmental Cluster (BSEC) library in order to get Indoor Air Quality (IAQ) readings.

Wi-Fi drivers
-------------

* Updated:

  * TCP/IP checksum offload to enable by default for the nRF70 Series.
  * Provision to change TX power ceilings using DTS file.

Libraries
=========

This section provides detailed lists of changes by :ref:`library <libraries>`.

* Added:

  * :ref:`nrf_security` library, relocated from the sdk-nrfxlib repository to the :file:`subsys/nrf_security` directory.
  * :ref:`network_core_monitor` library for monitoring the status of the nRF5340 processor's network core.

Debug libraries
---------------

* :ref:`cpu_load` library:

  * Updated by aligning the timer's configuration to the new nrfx API.

Binary libraries
----------------

* :ref:`lib_bt_ll_acs_nrf53_readme` library:

  * Added a limitation about the lack of support for the +20 dBm setting when :ref:`building the nRF5340 Audio application with the nRF21540 FEM support <nrf53_audio_app_adding_FEM_support>`.

* :ref:`liblwm2m_carrier_readme` library:

>   * Updated to 3.3.0.

Reviewer 4: Expected to bump to 3.3.1 today
https://github.com/nrfconnect/sdk-nrf/pull/12699

    See the :ref:`liblwm2m_carrier_changelog` for detailed information.

Bluetooth libraries and services
--------------------------------

> * :ref:`bt_fast_pair_readme` library:

Reviewer 3: @alstrzebonski, please also take a look at the Fast Pair library section in the NCS 2.5.0 release notes

************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.5.0`_ for the list of issues valid for the latest release.

.. _ncs_release_notes_250_changelog:

Changelog
*********

The following sections provide detailed lists of changes by component.

IDE and tool support
====================

|no_changes_yet_note|

Application development
=======================

This section provides detailed lists of changes to overarching SDK systems and components.

Build system
------------

* Removed the ``CONFIG_MCUBOOT_IMAGE_VERSION`` Kconfig option in favor of using a dedicated :ref:`application VERSION file <zephyr:app-version-details>` to set the version.
  You can alternatively set the version by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, but using a :file:`VERSION` file is the recommended approach.

* The |NCS| name and version is now displayed instead of the Zephyr version as the default boot banner when applications boot.
  This can be customized in user applications.

nRF Front-End Modules
---------------------

* Updated the name of the ``nrf21540_ek`` shield to ``nrf21540ek``.

> Working with nRF91 Series

Reviewer 5: Hi @peknis , regarding pan7 I found no good place to add this information. I am wondering if this can go to "working with nrf91 series" section. What do you think?

Author: Might be a good place for it.

* Added support for :ref:`nrf91_modem_trace_uart_snippet`.
  Snippet is used for nRF91 modem tracing with the UART backend for the following applications and samples:

  * :ref:`asset_tracker_v2`
  * :ref:`serial_lte_modem`
  * All samples that use nRF91 Series DK except for :ref:`slm_shell_sample`, :ref:`modem_trace_flash`, :ref:`modem_trace_backend_sample`.

  For samples where the UART trace backend is enabled by default, the configuration is added to the sample overlays and project configuration.

* The default board revision for nRF9160 DK has changed to v0.14.0.
  See :ref:`nrf9160_board_revisions` for more details.

Working with nRF52 Series
=========================

* :ref:`ug_nrf52_developing`:

  * Updated the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option to support MCUboot bootloader in the direct-xip mode, and added the related documentation.
    For details, see :ref:`mcuboot_direct_xip_mode`.

Working with nRF53 Series
=========================

|no_changes_yet_note|

Protocols
=========

This section provides detailed lists of changes by :ref:`protocol <protocols>`.
See `Samples`_ for lists of changes for the protocol-related samples.

> 

Reviewer 6: @peknis as discussed, apart from the information regarding Sidewalk landing pages that are already in the ncs documentation, adding the release notes
@ktaborowski @wojciechpluszynski  FYI
```suggestion

Amazon Sidewalk
---------------

Starting from 2.5.0 release, Amazon Sidewalk is a part of the nRF Connect SDK.

* Added: 

  * nRF53 production support for Amazon Sidewalk, including Bluetooth LE, LoRa, and FSK.
  * New distribution model (Amazon Sidewalk is now merged with nRF Connect SDK).

* Updated:

  * Suspended external flash in the Sidewalk mode to reduce power consumption. 
    External flash is only available in the DFU mode.
  * Merged Amazon Sidewalk libraries into a unified library.
  * Adjusted the sensor monitoring app to change to initialization state upon reconnecting.
    This setting prevents the cloud application from being stuck when it is disconnected for a long time.
  * Deactivated the Bluetooth LE GATT Client to optimize the configuration.

* Removed an experimental devcontainer for a better user experience.

```

* :ref:`west_sbom`:

  * Changed:

    * To reduce RAM usage, the script now runs the `Scancode-Toolkit`_ detector in a single process.
      This change slows down the licenses detector, because it is no longer executed simultaneously on all files.
    * SPDX License List database updated to version 3.22.

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``74c4d1c52fd51d07904b27a7aa9b2303e896a4e3``, with some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

|no_changes_yet_note|

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``a8b28f13c195a00bdf50f5c24092981124664ed9``, with some |NCS| specific additions.

> For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

Reviewer 2: ```suggestion
For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into |NCS| since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):
```

  * A page on :ref:`ug_wireless_coexistence` in :ref:`protocols`.
  * A page on :ref:`ug_sidewalk` in :ref:`protocols`.
  * Pages on :ref:`thread_device_types` and :ref:`thread_sed_ssed` to the :ref:`ug_thread` documentation.
  * A new page :ref:`ug_pmic`, containing :ref:`ug_npm1300_features` and :ref:`ug_npm1300_gs`.
  * A section about :ref:`nrf70_gs_shields_expansion_boards` in :ref:`nrf7002dk_nrf5340` user guide.
  * A page on :ref:`ug_nrf70_developing_scan_operation` in the :ref:`ug_nrf70_developing` user guide.
  * The :ref:`ug_bt_qualification` page in :ref:`protocols`.
  * A section on Wi-Fi in the :ref:`app_memory` page.
  * Own page for :ref:`bt_mesh_samples`.
  * :ref:`migration_2.5`.

* Updated:

  * :ref:`samples` with separate sections for :ref:`keys_samples` and :ref:`peripheral_samples`, which were previously listed in :ref:`other_samples`.
  * The :ref:`emds_readme` library documentation with :ref:`emds_readme_application_integration` section about the formula used to compute the required storage time at shutdown in a worst case scenario.

>   * The structure of the :ref:`nrf_modem_lib_readme` documentation.

Reviewer 7: Apart from structure change, Modem tracing with RTT section was added as part of the PR (https://github.com/nrfconnect/sdk-nrf/pull/11440/files). May be this could be added as a change log entry? https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/modem/nrf_modem_lib/nrf_modem_lib_trace.html#modem-tracing-with-rtt

Author: Done

    * LE Power Control request procedure that was introduced in |NCS| 2.2.0 as experimental is now supported.

  * Bluetooth Mesh:

    * Networked Lighting Controls (NLC), a set of Bluetooth Mesh profiles for standardized wireless lighting control.
    * Bluetooth Mesh 1.1 features introduced in nRF Connect SDK 2.4.0 as experimental are now supported.

      .. note::
         Samples built with Bluetooth Mesh 1.1 will be labeled as experimental when built.
         The experimental flag in the Zephyr code base could not be removed in time for the release unfortunately, but this is planned to be changed.

  * Sidewalk:

    * :ref:`ug_sidewalk` is now integrated as part of |NCS|, supporting nRF52840 and nRF5340 SoCs.

  * Wi-Fi:

    * Enhanced Wi-Fi scan API, providing optimizations and high user configurability for the Wi-Fi scanning functionality on nRF70 Series.
    * Network agnostic connectivity manager, allowing seamless use of IP-based protocols regardless of underlying transport (Cellular IoT, Wi-Fi).
    * :ref:`aws_iot`, :ref:`azure_iot_hub`, :ref:`download_sample`, :ref:`https_client`, :ref:`udp_sample`, and :ref:`wifi_zephyr_samples` are now supported on nRF7002 DK.

>     * New samples: :ref:`wifi_twt_sample`, `wifi_shutdown_sample`, and :ref:`wifi_wfa_qt_app_sample`

Reviewer 8: missing :ref: for the  `wifi_shutdown_sample`

Author: Thanks Ulf! Fixed.

  * An application core crash on nRF5340 targets with the factory data module enabled.
    The crash would happen after the OTA firmware update finishes and the image is confirmed.

See `Matter samples`_ for the list of changes for the Matter samples.

Matter fork
+++++++++++

The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, the ``v1.1.0.1`` tag.

The following is the most important change inherited from the upstream Matter:

* Added the :kconfig:option:`CONFIG_CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT` Kconfig option to manage watermark support.

* Updated:

  * The factory data guide with an additional rotating ID information.
  * Set onboarding code generation to be enabled by default if the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` Kconfig is set.

* Fixed RAM and ROM reports.

Zigbee
------

The Zigbee stack integrated with the |NCS| is not eligible for certification.
It should only be used for evaluation and prototyping, and should not be used in an end product.
A fixed, production ready version of the Zigbee stack will be part of the next release of the |NCS|.

> 

Author: Are you sure we want to add this? We have the following statement in the highlights:

* Deprecated:

  * With the introduction of Matter, all HomeKit customers are recommended to use Matter for new designs of smart home products.
    As a result, HomeKit Accessory Development Kit has been deprecated, and it will be removed in the next release of |NCS|.

> 

Author: Added as agreed offline.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

The following list includes the summary of the most relevant changes introduced in this release.

* Added :ref:`support <software_maturity>` for the following features:

  * Bluetooth® Low Energy:

    * LE Power Control request procedure that was introduced in |NCS| 2.2.0 as experimental is now supported.

  * Bluetooth Mesh:

    * Networked Lighting Controls (NLC), a set of Bluetooth Mesh profiles for standardized wireless lighting control.
    * Bluetooth Mesh 1.1 features introduced in nRF Connect SDK 2.4.0 as experimental are now supported.

      .. note::
         Samples built with Bluetooth Mesh 1.1 will be labeled as experimental when built.
         The experimental flag in the Zephyr code base could not be removed in time for the release unfortunately, but this is planned to be changed.

  * Sidewalk:

    * :ref:`ug_sidewalk` is now integrated as part of |NCS|, supporting nRF52840 and nRF5340 SoCs.

>   * Wi-F®i:

Author: Whoops ;)

    * Enhanced Wi-Fi scan API, providing optimizations and high user configurability for the Wi-Fi scanning functionality on nRF70 Series.
    * Network agnostic connectivity manager, allowing seamless use of IP-based protocols regardless of underlying transport (Cellular IoT, Wi-Fi).
    * :ref:`aws_iot`, :ref:`azure_iot_hub`, :ref:`download_sample`, :ref:`https_client`, :ref:`udp_sample`, and :ref:`wifi_zephyr_samples` are now supported on nRF7002 DK.

>     * New samples: :ref:`wifi_twt_sample`, :ref:`wifi_shutdown_sample`, and :ref:`wifi_wfa_qt_app_sample`

Reviewer 9: ```suggestion
    * New samples: :ref:`wifi_twt_sample`, :ref:`wifi_shutdown_sample`, and :ref:`wifi_wfa_qt_app_sample`.
```

  * The AT command ``#XSLMUART``.
    UART is now configured using only devicetree.

nRF5340 Audio
-------------

* Modified the entire application architecture for handling Bluetooth LE Audio.
  The following new modules have been added:

  * Management - This module handles scanning and advertising, in addition to general initialization, controller configuration, and transfer of DFU images.
    The new architecture makes it possible to make connections and handle periodic advertising sync independently of the Bluetooth LE Audio setup.
  * Stream - This module handles the setup and transfer of audio in the Bluetooth LE Audio context.
    The new architecture makes it possible to have more than one Bluetooth LE Audio role in one device.
  * Renderer - This module handles rendering, such as volume up and down.
  * Content Control - This module handles content control, such as play and pause.

* Added back the QDID number for the :ref:`lib_bt_ll_acs_nrf53_readme` to the documentation.
* Updated the :ref:`application documentation <nrf53_audio_app>` by splitting it into several pages.

> * Fixed a compatibility issue when using Google Pixel 7/7A.

Reviewer 10: ```suggestion

```
We still miss one config for compatibility with Google Pixel 7/7A. so we have to remove this point for now

Reviewer 10: This was discovered today

* :ref:`bluetooth_mesh_light` sample:

  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_light_lc` sample:

  * Added support for Composition Data Pages 1 and 2.
    Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.
  * Fixed an issue where the sample could return an invalid Light Lightness Status message if the transition time was evaluated to zero.
  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_light_dim` sample:

  * Added support for Composition Data Pages 1 and 2.
    Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.
  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_light_switch` sample:

  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_sensor_server` sample:

>   * Added support for Composition Data Pages 1 and 2.
>     Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.
>   * Added a getter for the :c:var:`bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range` sensor.

Reviewer 11: Could consider changing this to:
```suggestion
  * Added:
    
    * Support for Composition Data Pages 1 and 2, which has a dependency on Bluetooth mesh 1.1 support.
    * A getter for the :c:var:`bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range` sensor.
```

  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.
  * Fixed an issue where the :c:var:`bt_mesh_sensor_time_since_presence_detected` sensor could report an invalid value when the time delta would exceed the range of the characteristic.

> * Fixed an issue where some samples copied using the `nRF Connect for Visual Studio Code`_ extension would not compile due to relative paths in :file:`CMakeLists.txt`, which were referencing files outside of the applications folder.

Reviewer 11: This bullet is a general comment for mesh samples and should be moved up to right under the section, as the first entry.

* :ref:`http_application_update_sample` sample:

  * Updated credentials for the HTTPS connection.

* :ref:`http_modem_full_update_sample` sample:

  * Updated credentials for the HTTPS connection.

* :ref:`http_modem_delta_update_sample` sample:

  * Updated credentials for the HTTPS connection.

* :ref:`https_client` sample:

  * Updated the TF-M Mbed TLS overlay to fix an issue when connecting to the server.

* :ref:`nrf_cloud_rest_cell_pos_sample` sample:

  * Added:

    * The ``disable_response`` parameter to the :c:struct:`nrf_cloud_rest_location_request` structure.
      If set to true, no location data is returned to the device when the :c:func:`nrf_cloud_rest_location_get` function is called.
    * A Kconfig option :kconfig:option:`CONFIG_REST_CELL_LOCATION_SAMPLE_VERSION` for the sample version.
    * Reporting of device and connection info to the device shadow.

  * Updated the sample to print its version when started.

* :ref:`modem_shell_application` sample:

  * Added:

    * Support for controlling proprietary Power Saving Mode (PSM).
    * Support for accessing nRF Cloud services using CoAP through the :ref:`lib_nrf_cloud_coap` library.

>     * Support for GSM 7bit encoded hexadecimal string in SMS messages.

Reviewer 11: ```suggestion
    * Support for GSM 7-bit encoded hexadecimal string in SMS messages.
```

* :ref:`bluetooth_mesh_light_lc` sample:

  * Added support for Composition Data Pages 1 and 2.
    Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.
  * Fixed an issue where the sample could return an invalid Light Lightness Status message if the transition time was evaluated to zero.
  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_light_dim` sample:

  * Added support for Composition Data Pages 1 and 2.
    Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.
  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_light_switch` sample:

  * Removed support for the configuration with :ref:`CMSE enabled <app_boards_spe_nspe_cpuapp_ns>` for :ref:`zephyr:thingy53_nrf5340`.

* :ref:`bluetooth_mesh_sensor_server` sample:

  * Added:

  * Support for Composition Data Pages 1 and 2.
    Support for Composition Data Pages 1 and 2 has a dependency on Bluetooth mesh 1.1 support.

>   * A getter for the :c:var:`bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range` sensor.

Reviewer 11: Indent issue

* :ref:`nrf_cloud_rest_fota` sample:

  * Added reporting of device and connection info to the device shadow.

* :ref:`nrf_cloud_rest_device_message` sample:

  * Added:

    * A DTS overlay file for LEDs on the nRF9160 DK to be compatible with the :ref:`caf_leds`.
    * Header files for buttons and LEDs definition required by the :ref:`lib_caf` library.
    * An :file:`overlay-nrf_provisioning.conf` file to enable the :ref:`lib_nrf_provisioning` library.

  * Updated:

    * The sample to use the :ref:`lib_caf` library instead of the :ref:`dk_buttons_and_leds_readme` library.
    * The sample now displays an error message when it fails to send an alert to nRF Cloud.

* :ref:`udp` sample:

   * Updated the sample to use the Kconfig option :kconfig:option:`CONFIG_LTE_RAI_REQ` and socket options ``SO_RAI_*`` for Release Assistance Indication (RAI) functionality.

Thread samples
--------------

* Removed support for the ``nrf52833dk_nrf52833`` build target in the :ref:`ot_cli_sample`, :ref:`coap_client_sample`, and :ref:`coap_server_sample` samples.
* Updated the build target ``nrf52840dongle_nrf52840`` to use USB CDC ACM as serial transport as default.

>   Samples for this target can now be built without providing extra configuration arguments.

Reviewer 11: Switch the two bullets (updated before removed)?

    Previously, it was set incorrectly.

* :ref:`matter_template_sample`:

  * Removed support for the Thread, Wi-Fi, and software diagnostics clusters from the ZAP file.

Networking samples
------------------

* Added a new :ref:`udp_sample` sample that has support for Wi-Fi and LTE connectivity.
  The :ref:`udp` sample continues to serve as a low power example that sends UDP packets over LTE connection.

* :ref:`aws_iot` sample:

  * Added support for Wi-Fi and LTE connectivity through the connection manager API.
  * Updated by moving the sample from :file:`cellular/aws_iot` folder to :file:`net/aws_iot`.
    The documentation is now found in the :ref:`networking_samples` section.

* :ref:`azure_iot_hub` sample:

  * Added:

    * Support for Wi-Fi and LTE connectivity through the connection manager API.
    * Support for the nRF9161 development kit.
    * FOTA support using the :ref:`lib_azure_fota` library.

  * Updated by moving the sample from :file:`cellular/azure_iot_hub` folder to :file:`net/azure_iot_hub`.
    The documentation is now found in the :ref:`networking_samples` section.

* Removed Cellular: Azure FOTA sample.

>   FOTA using Azure IoT Hub is now demonstrated in the :ref:`azure_iot_hub` sample.

Reviewer 11: This should be moved further up, as the 2nd bullet in this section.

      This is due to binding with Generic Level state when receiving Generic Delta Set and Generic Move Set messages.
    * The :c:member:`bt_mesh_light_hue_srv_handlers.move_set` callback of the :ref:`bt_mesh_light_hue_srv_readme` model is only called for a continuous transition.
      All other transitions are now handled by the :c:member:`bt_mesh_light_hue_srv_handlers.set` callback.
    * The Hue Range state of the :ref:`bt_mesh_light_hue_srv_readme` model now allows :c:member:`bt_mesh_light_hsl_range.max` to be lower than :c:member:`bt_mesh_light_hsl_range.min`.

  * Fixed:

    * An issue where the :ref:`bt_mesh_dtt_srv_readme` model could not be found for models spanning multiple elements.
    * An issue where the :ref:`bt_mesh_sensor_srv_readme` model would add a corrupted marshalled sensor data into the Sensor Status message, because the fetched sensor value was outside the range.
      If the fetched sensor value is out of range, the marshalled sensor data for that sensor is not added to the Sensor Status message.

>   * Removed the ``bt_mesh_light_hue_srv_handlers.delta_set`` callback of the :ref:`bt_mesh_light_hue_srv_readme` model is removed and replaced with the :c:member:`bt_mesh_light_hue_srv_handlers.set` callback.

Reviewer 11: ```suggestion
  * Removed the ``bt_mesh_light_hue_srv_handlers.delta_set`` callback of the :ref:`bt_mesh_light_hue_srv_readme` model and replaced it with the :c:member:`bt_mesh_light_hue_srv_handlers.set` callback.
```

  * Aligned Kconfig option names in the :ref:`nrf_desktop_motion` implementation that generates motion from button presses.
    The Kconfig options defining used key IDs are prefixed with ``CONFIG_MOTION_BUTTONS_`` instead of ``CONFIG_MOTION_`` to ensure consistency with configuration of other implementations of the motion module.
  * The :ref:`nrf_desktop_ble_scan` no longer stops Bluetooth LE scanning when it receives :c:struct:`hid_report_event` related to a HID output report.
    Sending HID output report is triggered by a HID host.
    Scanning stop may lead to an edge case where the scanning is stopped, but there are no peripherals connected to the dongle.
  * Increased heap memory pool size (:kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE`) in nRF5340 DK configurations.
    This is done to prevent Event Manger out of memory (OOM) error.
  * Increased the stack size of a thread responsible for loading settings (:kconfig:option:`CONFIG_CAF_SETTINGS_LOADER_THREAD_STACK_SIZE`) to ``1200`` (default value) in the ``nrf52kbd_nrf52832`` configurations.

>     This is needed to prevent stack overflows on the initial boot right after programming the device.

Reviewer 3: Please add the following item:

```
* Aligned the documentation for the *NCS keyboard* and *NCS gaming mouse* Fast Pair debug models with the new configuration UI in the Google Nearby Console.
```

Reviewer 3: CC: @MarekPieta

Thingy:53: Matter weather station
---------------------------------

* Added support for the nRF7002 Wi-Fi expansion board.

Samples
=======

Bluetooth samples
-----------------

* :ref:`direct_test_mode` sample:

  * Added:

    * Support for the nRF52840 DK.
    * Experimental support for the HCI interface.

  * Updated:

    * Aligned timers' configurations to the new nrfx API.
    * Extracted the DTM radio API from the transport layer.
    * Added support for the radio fast ramp-up feature.
      This feature is enabled by default.

* :ref:`peripheral_hids_keyboard` sample:

  * Fixed an interoperability issue with iOS devices by setting the report IDs of HID input and output reports to zero.

* :ref:`fast_pair_input_device` sample:

  * Renamed the sample from Bluetooth: Fast Pair to :ref:`fast_pair_input_device` and moved it to the :file:`samples/bluetooth/fast_pair` folder.

>   * Added automatic switching to the Fast Pair not discoverable advertising mode with the hide UI indication instead of removing the Fast Pair advertising payload when all bond slots are taken.

Reviewer 3: Please add the following item after the "Renamed the sample ..." entry:
```
* Aligned the documentation for the Fast Pair debug model with the new configuration UI in the Google Nearby Console.
  Changed the Device Name from *NCS Fast Pair demo* to *NCS input device*.
```

      This allows the device to achieve two µA of current usage while in PSM sleep mode.
    * The sample to use the :kconfig:option:`CONFIG_LWM2M_UPDATE_PERIOD` Kconfig option to set the LwM2M update sending interval.

* :ref:`gnss_sample` sample:

  * Added support for nRF91x1 factory almanac.
    The new almanac file format also supports QZSS satellites.

* :ref:`nrf_cloud_rest_fota` sample:

  * Added reporting of device and connection info to the device shadow.

* :ref:`nrf_cloud_rest_device_message` sample:

  * Added:

    * A DTS overlay file for LEDs on the nRF9160 DK to be compatible with the :ref:`caf_leds`.
    * Header files for buttons and LEDs definition required by the :ref:`lib_caf` library.
    * An :file:`overlay-nrf_provisioning.conf` file to enable the :ref:`lib_nrf_provisioning` library.

  * Updated:

    * The sample to use the :ref:`lib_caf` library instead of the :ref:`dk_buttons_and_leds_readme` library.
    * The sample now displays an error message when it fails to send an alert to nRF Cloud.

* :ref:`udp` sample:

>    * Updated the sample to use the Kconfig option :kconfig:option:`CONFIG_LTE_RAI_REQ` and socket options ``SO_RAI_*`` for Release Assistance Indication (RAI) functionality.

Reviewer 12: ```suggestion
  * Updated:

    * The sample to use the Kconfig option :kconfig:option:`CONFIG_LTE_RAI_REQ` and socket options ``SO_RAI_NO_DATA``, ``SO_RAI_LAST``, and ``SO_RAI_ONGOING`` for Release Assistance Indication (RAI) functionality.
    * The documentation to showcase how to test the RAI functionality.
```

  * Socket option ``TLS_DTLS_HANDSHAKE_TIMEO`` to a new name value.
  * ``#XTCPSVR`` connection closure status and documentation.
  * ``#XRECVFROM`` to include the port of the peer.

* Removed:

  * DFU AT commands ``#XDFUGET``, ``#XDFUSIZE`` and ``#XDFURUN`` because they were not usable without a custom application in the target (nRF52 Series) device.
  * Support for bootloader FOTA update because it is not needed for Serial LTE modem.
  * Operations to read or erase the MCUboot secondary slot from the ``#XFOTA`` AT command because the application update process overwrites the slot in any case.
  * AT commands ``#XCELLPOS`` and ``#XWIFIPOS``.
    They are replaced by the ``#XNRFCLOUDPOS`` command that allows to combine cellular and Wi-Fi data to determine the device location.
  * The AT commands ``#XAGPS`` and ``#XPGPS``.
    Their functionality is merged into the ``#XGPS`` AT command that now allows using A-GNSS and P-GPS at the same time.
  * The AT command ``#XSLMUART``.
    UART is now configured using only devicetree.

> 

Author: Added.

  * Wi-Fi®:

    * Enhanced Wi-Fi scan API, providing optimizations and high user configurability for the Wi-Fi scanning functionality on nRF70 Series.
    * Network agnostic connectivity manager, allowing seamless use of IP-based protocols regardless of underlying transport (Cellular IoT, Wi-Fi).
    * :ref:`aws_iot`, :ref:`azure_iot_hub`, :ref:`download_sample`, :ref:`https_client`, :ref:`udp_sample`, and :ref:`wifi_zephyr_samples` are now supported on nRF7002 DK.
    * New samples: :ref:`wifi_twt_sample`, :ref:`wifi_shutdown_sample`, and :ref:`wifi_wfa_qt_app_sample`.

  * Cellular IoT:

    * Software SIM API, allowing the usage of Software SIM-based solutions to reduce energy consumption associated with physical SIMs and allowing for more compact hardware designs.
    * New samples: :ref:`battery` and :ref:`nrf_provisioning_sample`.

  * Power Management (nPM1300):

    * nPM1300 features introduced in |NCS| 2.4.0 as experimental are now supported.
    * New features: LEDs, ship, hibernate and reset, more charger configurations, including JEITA and trickle charging, watchdog and event handling.

  * DFU:

>     * :ref:`Direct XiP mode for MCUboot <ug_nrf52_developing_ble_fota_mcuboot_direct-xip_mode>`, reducing downtime during the DFU process.

Reviewer 3: ```suggestion
    * :ref:`Direct XiP mode for MCUboot <ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode>`, reducing downtime during the DFU process.
```

  This can be customized in user applications.

nRF Front-End Modules
---------------------

* Updated the name of the ``nrf21540_ek`` shield to ``nrf21540ek``.

Working with nRF91 Series
=========================

* Added support for :ref:`nrf91_modem_trace_uart_snippet`.
  Snippet is used for nRF91 modem tracing with the UART backend for the following applications and samples:

  * :ref:`asset_tracker_v2`
  * :ref:`serial_lte_modem`
  * All samples that use nRF91 Series DK except for :ref:`slm_shell_sample`, :ref:`modem_trace_flash`, :ref:`modem_trace_backend_sample`.

  For samples where the UART trace backend is enabled by default, the configuration is added to the sample overlays and project configuration.

* The default board revision for nRF9160 DK has changed to v0.14.0.
  See :ref:`nrf9160_board_revisions` for more details.

Working with nRF52 Series
=========================

* :ref:`ug_nrf52_developing`:

  * Updated the :kconfig:option:`CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU` Kconfig option to support MCUboot bootloader in the direct-xip mode, and added the related documentation.

>     For details, see :ref:`mcuboot_direct_xip_mode`.

Reviewer 3: ```suggestion
    For details, see :ref:`ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode`.
```

    The size of the span partitions now reflects the memory space taken from the start of the first of its elements to the end of the last, not just the sum of the sizes of the included partitions.
  * Fixed a bug where the ``align`` spec was deleted.
    This would happen in cases where two ``placement`` specs were identical.
    When disambiguating one of them, the ``align`` spec was not preserved.

* :ref:`west_sbom`:

  * Updated:

    * To reduce RAM usage, the script now runs the `Scancode-Toolkit`_ detector in a single process.
      This change slows down the licenses detector, because it is no longer executed simultaneously on all files.
    * SPDX License List database updated to version 3.22.

MCUboot
=======

The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``11ecbf639d826c084973beed709a63d51d9b684e``, with some |NCS| specific additions.

The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

> * Added a version check for network core when downgrade prevention is enabled.

Author: Looks like it is not going in.
